input/parser_test.go

```diff
@@ -1,105 +1,214 @@
 package input
 
-import "testing"
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
 
-/*
- * RFC5424 parser tests
- */
-
-func Test_SuccessfulRFC5424Parsing(t *testing.T) {
-	p := NewRFC5424Parser()
+func Test_Formats(t *testing.T) {
+	var p *Parser
+	mismatched := func(rtrnd string, intnd string, intndA string) {
+		if intndA != "" {
+			t.Fatalf("Parser format %v does not match the intended format %v.\n", rtrnd, intnd)
+		} else {
```

Comment:
Go style dictates that there is no need for an `else` clause here. Just call the second `Fatalf` directly.